.travis.yml

```diff
@@ -42,3 +52,76 @@ deploy:
     - dist/pomerium-windows-amd64.md5
     - dist/pomerium-windows-amd64.sha256
   skip_cleanup: true
+
+docker_setup: &docker_setup
+  - curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+  - sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+  - sudo apt-get update
+  - sudo apt-get -y -o Dpkg::Options::="--force-confnew" install docker-ce
+  - docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+
+docker_login: &docker_login
+  - echo ${DOCKER_PASSWORD} | docker login -u pomeriumci --password-stdin
```

Comment:
>Would docker login -u pomeriumci -p ${DOCKER_PASSWORD} work? I don't know how likely a mangled "echo" would be but it was just a though.

Old school standard practice since it can't be easily seen if you're another user on the machine.  Passing the parameter on the command line is visible to anything on the machine.

>And maybe even set an envar ${DOCKER_USERNAME} in case we need/want to switch the user at some point?

Yes good call.